zeeguu_api/api/user_languages.py

```diff
@@ -0,0 +1,137 @@
+import flask
+import zeeguu
+from flask import request
+from zeeguu.model.language import Language
+from zeeguu.model.user_language import UserLanguage
+
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+USER_LANGUAGES = "user_languages"
+ADD_USER_LANGUAGE = "user_languages/add"
+DELETE_USER_LANGUAGE = "user_languages/delete"
+INTERESTING_LANGUAGES = "user_languages/interesting"
+READING_LANGUAGES = "user_languages/reading"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{ADD_USER_LANGUAGE}", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def add_user_language():
+    """
+    This endpoint is for adding a user language.
+    It tries to find the user_language, and otherwise create it.
+    After that set it to reading_news so it appears in the reader.
+
+    :return: "OK" in case of success
+    """
+    language_id = int(request.form.get('language_id', ''))
+
```

Comment:
would be nice to rename to `modify_user_language`

and allow also the setting of the other params:

    language_declared_level = int(request.form.get('declared_level', ''))
    doing_exercises = ...